test/replication.test.js

```diff
@@ -1804,3 +1804,105 @@ describe('Replication / Change APIs', function() {
     return getPropValue(list, 'id');
   }
 });
+
+describe('Replication / Change APIs with multi tenancy', function() {
+  this.timeout(10000);
+  var dataSource, SourceModelWithTenant, TargetModelWithTenant;
+  var useSinceFilter;
+  var tid = 0; // per-test unique id used e.g. to build unique model names
+
+  beforeEach(function() {
+    tid++;
+    useSinceFilter = false;
+    var test = this;
+
+    dataSource = this.dataSource = loopback.createDataSource({
+      connector: loopback.Memory
+    });
+    SourceModelWithTenant = this.SourceModelWithTenant = PersistedModel.extend(
```

Comment:
We are moving away from global registry of models in favour of local per-app registries. That way there is no need for `tid`.

```js
app = loopback({localRegistry: true, loadBuiltinModels: true});
app.dataSource('db', {connector: 'memory'});

SourceModelWithTenant = app.registry.createModel({
  name: 'SourceModelWithTenant',
  properties: { /*...*/ },
  trackChanges: true
});
app.model(SourceModelWithTenant, { dataSource: 'db' });

// etc.
```

Also `tenantProperty` setting is not needed here.